### ISSUE TITLE
`cabal init` should warn if package name matches any known existing package

### ISSUE THREAD

**Issue by lspitzner** [haskell/cabal#2372]
and it should suggest a different default if it is an existing package. consider

```
> mkdir lens
> cd lens
> cabal init
Package name? [default: lens]:     -- meh
```

because if you now add a (transitive) dependency on the "actual" `lens` package, stuff won't work.

**Comment by lspitzner** [comment]
this.labels += enhancement;

**Comment by cfoch** [comment]
Hello. I've written some code that if you choose a package name of an already installed package it will ask you if you want to continue. Otherwise, you can choose another name. 
I'm not sure if the way I am using to check if a package is already installed is the best way, though. I would like someone to review my code, please.
https://github.com/cfoch/cabal/tree/initChooseOtherName

**Comment by dcoutts** [comment]
Thanks for this. Main comment is that we should be looking in the source package index rather than the installed package index, for name clashes.

**Comment by grayjay** [comment]
This was probably fixed by #2436.  I just tried running `cabal init` with master:

```
$ mkdir lens

$ cd lens/

$ cabal init
Package name? lens
This package name is already used by another package on hackage. Do you want to choose a different name? [default: y] 
Package name? my-package
Package version? [default: 0.1.0.0]
```
